src/core/thread/network_data_leader.cpp

```diff
@@ -518,6 +519,7 @@ ThreadError Leader::RemoveCommissioningData(void)
         if (cur->GetType() == NetworkDataTlv::kTypeCommissioningData)
         {
             Remove(reinterpret_cast<uint8_t *>(cur), sizeof(NetworkDataTlv) + cur->GetLength());
+            mLength -= (sizeof(NetworkDataTlv) + cur->GetLength());
```

Comment:
I thought `NetworkData::Remove()` decreases `mLength` appropriately.  Why do we need this here?